lib/webauth.js

```diff
@@ -185,24 +185,39 @@ module.exports = function(authManager) {
         const authPluginSession = getAuthPluginSession(req, pluginID, {});
         req[`${UNP.APP_NAME}Data`].plugin.services = 
           authServiceHandleMaps[pluginID];
-        const wasAuthenticated = handler.getStatus(authPluginSession).authenticated;
-        let handlerResult;
         
-        //dont allow refresh if zlux cant find id or overall session expired
-        if (type === SESSION_ACTION_TYPE_REFRESH
-            && authManager.sessionTimeoutMs !== TIMEOUT_VALUE_NO_EXPIRE
-            && (!timeout || timeout < Date.now())) {
-          req.session.zlux = undefined;
-          handlerResult = {success:false, reason:REASON_ZLUX_SESSION_EXPIRE};
+        const authHandlerStatus = handler.getStatus(authPluginSession);
+        const wasAuthenticated = authHandlerStatus.authenticated;
+        const hasGetCapabilities = !!((typeof handler.getCapabilities) === 'function');
+        let handlerResult;
+
+        /*
+          The logic for performing refreshStatus()/authenticate() calls has been wrapped in a check for the presence of
+          the getCapabilities() function because functions that do not include getCapabilities() likely dont support refreshing (or haven't been maintained).  
+          If a service calls refreshStatus on an auth handler that does not support it,
+          { success: true } is returned so that the session may expire naturally and users are not forced to log in every time they refresh.
+        */
+        if(hasGetCapabilities){
+          if (type === SESSION_ACTION_TYPE_REFRESH
+              && authManager.sessionTimeoutMs !== TIMEOUT_VALUE_NO_EXPIRE
+              && (!timeout || timeout < Date.now())) {
+            req.session.zlux = undefined;
```

Comment:
This should not be inside of hasGetCapabilities - this is checking if the server itself has timed out, not if an auth plugin has.

Consider: 
if (this stuff) { 
REASON_ZLUX_SESSION_EXPIRE
} else if (hasGetCapabilities && canRefresh) { 
do refresh
} else {
success = true
}